lib/pure/uri.nim

```diff
@@ -277,6 +278,12 @@ func resetUri(uri: var Uri) =
     else:
       f = false
 
+func removeUnsafeBytesFromUri(uri: string): string =
+  var u = uri
+  for b in unsafeUrlBytesToRemove:
+    u = replace(u, b)
```

Comment:
inefficient, this allocates more than it needs to.
use `multiReplace`

Reply:
Would something like this work for a `removeInplace`?

```nim
func removeInPlace(result: var string, chars: set[char]) =
  ## Removes all characters from `chars` from string `s`
  ## (in-place).
  runnableExamples:
    var userInput = "line\t1\nline\t2"
    removeInPlace(u, {'\n', '\t'})
    doAssert userInput == "line1line2"

  if result.len == 0: return
  var i = 0
  while i < result.len:
    if result[i] in chars: result.delete(i, i)
    i += 1
```